firefox-ios/firefox-ios-tests/Tests/ClientTests/TabTray/TabsPanelStateTests.swift

```diff
@@ -34,9 +44,85 @@ final class TabPanelStateTests: XCTestCase {
+                                              inactiveTabs: [InactiveTabsModel](),
+                                              isInactiveTabsExpanded: false,
+                                              shouldScrollToTab: true)
+        try testTabsState_scrollToIndexHelper(actionType: .refreshTabs, tabDisplayModel: tabDisplayModel)
+    }
+
+    func testTabsState_scrollToIndexHelper(actionType: TabPanelMiddlewareActionType,
+                                           tabDisplayModel: TabDisplayModel,
+                                           file: StaticString = #file,
+                                           line: UInt = #line) throws {
+        let initialState = createInitialState()
+        XCTAssertNil(initialState.scrollToIndex, file: file, line: line)
+        let selectedIndex = try XCTUnwrap(tabDisplayModel.tabs.firstIndex(where: \.isSelected), file: file, line: line)
+        let reducer = tabsPanelReducer()
+        let action = TabPanelMiddlewareAction(
+            tabDisplayModel: tabDisplayModel,
+            windowUUID: .XCTestDefaultUUID,
+            actionType: actionType
+        )
         let newState = reducer(initialState, action)
-        XCTAssertFalse(newState.tabs.isEmpty)
+        XCTAssertEqual(newState.scrollToIndex, selectedIndex, "Expected index was: \(selectedIndex)", file: file, line: line)
+    }
+
+    func testTabsState_refreshInactiveTabs() throws {
+        let initialState = createInitialState()
+        XCTAssertTrue(initialState.inactiveTabs.isEmpty)
+        let reducer = tabsPanelReducer()
+        let tabs = createTabs()
+        let inactiveTabs = createInactiveTabs()
+        let tabDisplayModel = TabDisplayModel(isPrivateMode: false,
+                                              tabs: tabs,
+                                              normalTabsCount: "\(tabs.count)",
+                                              inactiveTabs: [InactiveTabsModel](),
+                                              isInactiveTabsExpanded: false,
+                                              shouldScrollToTab: false)
+        let action = TabPanelMiddlewareAction(
+            tabDisplayModel: tabDisplayModel,
+            inactiveTabModels: inactiveTabs,
+            windowUUID: .XCTestDefaultUUID,
+            actionType: TabPanelMiddlewareActionType.refreshInactiveTabs
+        )
+        let newState = reducer(initialState, action)
+        XCTAssertEqual(newState.inactiveTabs, inactiveTabs, "Expected inactive tabs were: \(inactiveTabs)")
+    }
+
+    func testTabsState_showToast() {
+        for toastType in toastTypes() {
+            let initialState = createInitialState()
+            XCTAssertNil(initialState.toastType)
+            let reducer = tabsPanelReducer()
+            let action = TabPanelMiddlewareAction(
+                toastType: toastType,
+                windowUUID: .XCTestDefaultUUID,
+                actionType: TabPanelMiddlewareActionType.showToast
```

Comment:
It seems that this is handled in `TabTrayState` and not in `TabsPanelState` now, so we don't need to add these tests here. As part of this ticket, we would like to increase coverage for actions that update the tabs panel state. 

There are still some areas that we can improve on despite the recent additions of tests to this class, see lack of coverage in the following screenshot:
![image](https://github.com/user-attachments/assets/be8655d2-f32b-4251-9e30-67806b822126)

Therefore, the tests you wrote for `refreshInactiveTabs` and `refreshTabs` and `willAppearTabPanel` are still relevant and helpful! Let me know if you have further question and can give another review when this is updated.